File: 2023/06/ts/main.ts
import type { SolutionOptions } from 'src/options.ts';
import { run } from 'src/run.ts';

export const options: SolutionOptions = {
   hasAlternate: false,
   hasIo: false,
};

// a is 1 so we dont need it
// (-b +- sqrt(b**2 + 4 * a * c)) / 2 * a
function ohnomath(b: number, c: number, pepsilon = 0.001) {
   const min = Math.floor((b + Math.sqrt(b * b - 4 * c)) / 2 - pepsilon);
   const max = Math.ceil((b - Math.sqrt(b * b - 4 * c)) / 2 + pepsilon);
   return min - max + 1;
}

export function part1(input: string, _isTest: boolean): string {
   const td = input.split('\n').map((str) => str.split(':')[1].trim().split(/\s+/).map(Number));
   let res = 1;
   for (let i = 0; i < td[0].length; i++) res *= ohnomath(td[0][i], td[1][i]);
   return res.toString();
}

export function part2(input: string, _isTest: boolean): string {
   const td = input
      .split('\n')
      .map((str) => Number(str.split(':')[1].trim().split(/\s+/).join('')));
   return ohnomath(td[0], td[1]).toString();
}

if (import.meta.main) {
   run(Deno.args, part1, part2, options);
}

File: 2023/06/csharp/Main.cs

using Core;

namespace Year2023
{
   public class Day06
   {
      static SolutionOptions OPTIONS = new(
      )
      {
         HasAlternate = false,
         HasIO = false,
      };

      static long OhnoMath(long b, long c, double pepsilon = 0.001)
      {
         double min = Math.Floor((b + Math.Sqrt(b * b - 4 * c)) / 2 - pepsilon);
         double max = Math.Ceiling((b - Math.Sqrt(b * b - 4 * c)) / 2 + pepsilon);
         return (long)(min - max + 1);
      }

      static string Part1(string input, bool isTest = false)
      {
         long[][] td = input.Split('\n').Select(x => x.Split(':')[1].Trim().Split(' ').Where(x => x != "").Select(long.Parse).ToArray()).ToArray();
         long res = 1;
         for (int i = 0; i < td[0].Length; i++) res *= OhnoMath(td[0][i], td[1][i]);
         return res.ToString();
      }

      static string Part2(string input, bool isTest = false)
      {
         long[] td = input.Split('\n').Select(x => long.Parse(string.Join("", x.Split(':')[1].Split(' ')))).ToArray();
         return OhnoMath(td[0], td[1]).ToString();
      }

      static void Main(string[] args)
      {
         Run.Execute(args, Part1, Part2, OPTIONS);
      }
   }
}